home/utils.py

```diff
@@ -445,163 +349,60 @@ def style_sheet(wb: Workbook, sheet: Worksheet) -> Tuple[Workbook, Worksheet]:
-        str(page.body),
-        page.whatsapp_title,
-        messenging_platform_messages["whatsapp"][0]["msg"]
-        if len(messenging_platform_messages["whatsapp"]) == 1
-        else "",
-        page.messenger_title,
-        messenging_platform_messages["messenger"][0]["msg"]
-        if len(messenging_platform_messages["messenger"]) == 1
-        else "",
-        page.viber_title,
-        messenging_platform_messages["viber"][0]["msg"]
-        if len(messenging_platform_messages["viber"]) == 1
-        else "",
-        str(page.translation_key),
-        format_list_from_query_set(page.tags.all()),
-        format_list_from_query_set(page.quick_replies.all()),
-        format_list_from_query_set(page.triggers.all()),
-        str(page.locale),
-        get_next_prompts(messenging_platform_messages, 0),
-        get_image_link(messenging_platform_messages, 0),
-        get_doc_link(messenging_platform_messages, 0),
-        get_media_link(messenging_platform_messages, 0),
-        format_list_from_query_set(get_related_pages(page)),
-    ]
-    if most_messages == 0:
-        # if a page has no content
-        basic_row[5] = 0
-        return [basic_row]
-
-    for message_index in range(most_messages):
-        if message_index == 0:
-            row = basic_row
-        else:
-            row = padding + [""] * len(FIELDNAMES)
-            row[21] = get_next_prompts(messenging_platform_messages, message_index)
-            row[22] = get_image_link(messenging_platform_messages, message_index)
-            row[23] = get_doc_link(messenging_platform_messages, message_index)
-            row[24] = get_media_link(messenging_platform_messages, message_index)
-        row[5] = message_index + 1
-
-        if message_index < len(messenging_platform_messages["whatsapp"]):
-            row[11] = messenging_platform_messages["whatsapp"][message_index]["msg"]
-
-        if message_index < len(messenging_platform_messages["messenger"]):
-            row[13] = messenging_platform_messages["messenger"][message_index]["msg"]
-
-        if message_index < len(messenging_platform_messages["viber"]):
-            row[15] = messenging_platform_messages["viber"][message_index]["msg"]
-        rows.append(row)
+    base_row = ContentSheetRow(page, structure_string)
+    rows.append(base_row.get_row())
+    if base_row.tuple_of_extra_rows:
+        for extra_row in base_row.tuple_of_extra_rows:
+            rows.append(extra_row.get_row())
     return rows
 
 
 def add_children(
-    temp_sheet: List[list], children: PageQuerySet, queryset: PageQuerySet
+    temp_sheet: List[list],
+    children: PageQuerySet,
+    queryset: PageQuerySet,
+    parent_structure: str,
 ) -> List[list]:
     """Recursive function that traverses the children of a page with a depth first search algorithm"""
-    for child in children:
+    for index, child in enumerate(children, 1):
         content_page = ContentPage.objects.filter(id=child.id).first()
```

Comment:
if this doesn't exist will there be an error?

Reply:
`add_children` gets called in 2 places, both where `if content_page.has_children` is used so there will be a list of children only if there are children, so, I dont think it will ever not exist? but I am not sure